pkg/client/build.go

```diff
@@ -799,22 +802,22 @@ func (c *Client) getBuilder(img imgutil.Image) (*builder.Builder, error) {
 	return bldr, nil
 }
 
-func (c *Client) validateRunImage(context context.Context, name string, opts image.FetchOptions, expectedStack string) (imgutil.Image, error) {
+func (c *Client) validateRunImage(context context.Context, name string, opts image.FetchOptions, expectedStack string) (builderImage imgutil.Image, warnings []string, err error) {
 	if name == "" {
-		return nil, errors.New("run image must be specified")
+		return nil, nil, errors.New("run image must be specified")
 	}
 	img, err := c.imageFetcher.Fetch(context, name, opts)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	stackID, err := img.Label("io.buildpacks.stack.id")
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if stackID != expectedStack {
-		return nil, fmt.Errorf("run-image stack id '%s' does not match builder stack '%s'", stackID, expectedStack)
+		warnings = append(warnings, "deprecated usage of stack")
 	}
-	return img, nil
+	return builderImage, warnings, nil
```

Comment:
`builderImage` (but it should really be called `runImage`) is never getting set, so the returned value is nil (that is what is causing the nil exception). You need to set the image to the result of calling `c.imageFetcher.Fetch` on line 809.